hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java

```diff
@@ -76,34 +80,50 @@ public static FHIRJsonMessage parseJson2FHIRMsg(final JSONObject jsonObj) {
 			String key = hs.getKey();
 			Object value = hs.getValue();
 			if (value instanceof String) {
-				if (key.equals(FHIR_JSONMESSAGE_RESOURCETYPE)) {
-					fhirJsonMsg.setResourceType(value.toString());
-				} else if (key.equals(FHIR_JSONMESSAGE_STATUS)) {
-					fhirJsonMsg.setStatus(value.toString());
-				} else {
-					logger.error("This is not an valid FHIR message!");
-					return null;
-				}
+				parseJsonString(fhirJsonMsg, key, value);
```

Comment:
Business logic in new method is not same as old logic.
'return null' is ignored.